Tests/MapboxMapsTests/Foundation/Mocks/MockStyleManager.swift

```diff
@@ -0,0 +1,343 @@
+import Foundation
+@testable import MapboxMaps
+@_implementationOnly import MapboxCommon_Private
+
+final class MockStyleManager: StyleManagerType, Stubbable {
+
+    func asStyleManager() -> StyleManager {
+        fatalError()
+    }
+
+    @Stubbed var stubStyleURI: String!
+    func getStyleURI() -> String {
+        stubStyleURI ?? ""
+    }
+
+    func setStyleURIForUri(_ uri: String) {
+        stubStyleURI = uri
+    }
+
+    @Stubbed var stubStyleJSON: String!
+    func getStyleJSON() -> String {
+        stubStyleJSON
+    }
+
+    func setStyleJSONForJson(_ json: String) {
+        stubStyleJSON = json
+    }
+
+    @Stubbed var stubDefaultCamera: MapboxCoreMaps.CameraOptions!
+    func getStyleDefaultCamera() -> MapboxCoreMaps.CameraOptions {
+        stubDefaultCamera
+    }
+
+    @Stubbed var stubStyleTransition: MapboxCoreMaps.TransitionOptions!
+    func getStyleTransition() -> MapboxCoreMaps.TransitionOptions {
+        stubStyleTransition
+    }
+
+    func setStyleTransitionFor(_ transitionOptions: MapboxCoreMaps.TransitionOptions) {
+        stubStyleTransition = transitionOptions
+    }
+
+    // MARK: Style Layers
+
+    @Stubbed var stubStyleLayers: [MapboxCoreMaps.StyleObjectInfo] = .random(withLength: 3) {
+        MapboxCoreMaps.StyleObjectInfo(
+            id: .randomAlphanumeric(withLength: 12),
+            type: MapboxMaps.LayerType.random().rawValue)
+    }
+    func styleLayerExists(forLayerId layerId: String) -> Bool {
+        stubStyleLayers.contains(where: { $0.id == layerId })
+    }
```

Comment:
This doesn't record parameters like most stubs. A standalone Stub would be better here.

Also, the default return value seems very specific. I would recommend moving it into the test that needs it.